src/dialer/index.js

```diff
@@ -190,6 +202,45 @@ class Dialer {
     log('token %d released', token)
     this.tokens.push(token)
   }
+
+  /**
+   * Resolve multiaddr recursively.
+   *
+   * @param {Multiaddr} ma
+   * @returns {Promise<Array<Multiaddr>>}
+   */
+  async _resolve (ma) {
+    // TODO: recursive logic should live in multiaddr once dns4/dns6 support is in place
+    // Now only supporting resolve for dnsaddr
+    const resolvableProto = ma.protoNames().includes('dnsaddr')
```

Comment:
We could also just default to using resolvers for the other resolvable addresses that just return the existing multiaddr. This would give flexibility to anyone that needs to actually change the behavior.

Reply:
I agree that makes more sense! I will add those defaults

Reply:
We cannot do this change as dns4/dns6 would recursively resolve themselves.